{
    "timestamp": "2025-04-12 03:43:10",
    "client_ip": "127.0.0.1",
    "method": "POST",
    "url": "http://localhost:8000/posts/",
    "headers": {
        "host": "localhost:8000",
        "connection": "keep-alive",
        "content-length": "126",
        "sec-ch-ua-platform": "\"Windows\"",
        "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/132.0.0.0 YaBrowser/25.2.0.0 Safari/537.36",
        "accept": "application/json",
        "sec-ch-ua": "\"Not A(Brand\";v=\"8\", \"Chromium\";v=\"132\", \"YaBrowser\";v=\"25.2\", \"Yowser\";v=\"2.5\"",
        "content-type": "application/json",
        "sec-ch-ua-mobile": "?0",
        "origin": "http://localhost:8000",
        "sec-fetch-site": "same-origin",
        "sec-fetch-mode": "cors",
        "sec-fetch-dest": "empty",
        "referer": "http://localhost:8000/docs",
        "accept-encoding": "gzip, deflate, br, zstd",
        "accept-language": "ru,en;q=0.9"
    },
    "error_type": "IntegrityError",
    "error_message": "(psycopg2.errors.ForeignKeyViolation) insert or update on table \"post_table\" violates foreign key constraint \"fk_post_post_type\"\nDETAIL:  Key (post_type_id)=(0) is not present in table \"post_type_table\".\n\n[SQL: INSERT INTO post_table (post_id, content, child_id, user_id, media_link, creation_date, views_count, post_type_id) VALUES (%(post_id)s, %(content)s, %(child_id)s, %(user_id)s, %(media_link)s, %(creation_date)s, %(views_count)s, %(post_type_id)s)]\n[parameters: {'post_id': 4, 'content': '\u041f\u0440\u0438\u0432\u0435\u0435\u0435\u0435\u0435\u0435\u0435\u0442', 'child_id': None, 'user_id': 0, 'media_link': 'string', 'creation_date': datetime.date(2025, 4, 12), 'views_count': 0, 'post_type_id': 0}]\n(Background on this error at: https://sqlalche.me/e/20/gkpj)",
    "traceback": "Traceback (most recent call last):\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\engine\\base.py\", line 1964, in _exec_single_context\n    self.dialect.do_execute(\n    ~~~~~~~~~~~~~~~~~~~~~~~^\n        cursor, str_statement, effective_parameters, context\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\engine\\default.py\", line 945, in do_execute\n    cursor.execute(statement, parameters)\n    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^\npsycopg2.errors.ForeignKeyViolation: insert or update on table \"post_table\" violates foreign key constraint \"fk_post_post_type\"\nDETAIL:  Key (post_type_id)=(0) is not present in table \"post_type_table\".\n\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\starlette\\_exception_handler.py\", line 42, in wrapped_app\n    await app(scope, receive, sender)\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\starlette\\routing.py\", line 73, in app\n    response = await f(request)\n               ^^^^^^^^^^^^^^^^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\fastapi\\routing.py\", line 301, in app\n    raw_response = await run_endpoint_function(\n                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    ...<3 lines>...\n    )\n    ^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\fastapi\\routing.py\", line 214, in run_endpoint_function\n    return await run_in_threadpool(dependant.call, **values)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\starlette\\concurrency.py\", line 37, in run_in_threadpool\n    return await anyio.to_thread.run_sync(func)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\anyio\\to_thread.py\", line 56, in run_sync\n    return await get_async_backend().run_sync_in_worker_thread(\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n        func, args, abandon_on_cancel=abandon_on_cancel, limiter=limiter\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\anyio\\_backends\\_asyncio.py\", line 2470, in run_sync_in_worker_thread\n    return await future\n           ^^^^^^^^^^^^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\anyio\\_backends\\_asyncio.py\", line 967, in run\n    result = context.run(func, *args)\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\app\\routes.py\", line 67, in create_post\n    return PostService.create_post(db=db, post_data=post)\n           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\app\\services\\post_service.py\", line 35, in create_post\n    db.commit()\n    ~~~~~~~~~^^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\orm\\session.py\", line 2032, in commit\n    trans.commit(_to_root=True)\n    ~~~~~~~~~~~~^^^^^^^^^^^^^^^\n  File \"<string>\", line 2, in commit\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\orm\\state_changes.py\", line 139, in _go\n    ret_value = fn(self, *arg, **kw)\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\orm\\session.py\", line 1313, in commit\n    self._prepare_impl()\n    ~~~~~~~~~~~~~~~~~~^^\n  File \"<string>\", line 2, in _prepare_impl\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\orm\\state_changes.py\", line 139, in _go\n    ret_value = fn(self, *arg, **kw)\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\orm\\session.py\", line 1288, in _prepare_impl\n    self.session.flush()\n    ~~~~~~~~~~~~~~~~~~^^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\orm\\session.py\", line 4353, in flush\n    self._flush(objects)\n    ~~~~~~~~~~~^^^^^^^^^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\orm\\session.py\", line 4488, in _flush\n    with util.safe_reraise():\n         ~~~~~~~~~~~~~~~~~^^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\util\\langhelpers.py\", line 146, in __exit__\n    raise exc_value.with_traceback(exc_tb)\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\orm\\session.py\", line 4449, in _flush\n    flush_context.execute()\n    ~~~~~~~~~~~~~~~~~~~~~^^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\orm\\unitofwork.py\", line 466, in execute\n    rec.execute(self)\n    ~~~~~~~~~~~^^^^^^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\orm\\unitofwork.py\", line 642, in execute\n    util.preloaded.orm_persistence.save_obj(\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^\n        self.mapper,\n        ^^^^^^^^^^^^\n        uow.states_for_mapper_hierarchy(self.mapper, False, False),\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n        uow,\n        ^^^^\n    )\n    ^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\orm\\persistence.py\", line 93, in save_obj\n    _emit_insert_statements(\n    ~~~~~~~~~~~~~~~~~~~~~~~^\n        base_mapper,\n        ^^^^^^^^^^^^\n    ...<3 lines>...\n        insert,\n        ^^^^^^^\n    )\n    ^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\orm\\persistence.py\", line 1048, in _emit_insert_statements\n    result = connection.execute(\n        statement, multiparams, execution_options=execution_options\n    )\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\engine\\base.py\", line 1416, in execute\n    return meth(\n        self,\n        distilled_parameters,\n        execution_options or NO_OPTIONS,\n    )\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\sql\\elements.py\", line 523, in _execute_on_connection\n    return connection._execute_clauseelement(\n           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^\n        self, distilled_params, execution_options\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\engine\\base.py\", line 1638, in _execute_clauseelement\n    ret = self._execute_context(\n        dialect,\n    ...<8 lines>...\n        cache_hit=cache_hit,\n    )\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\engine\\base.py\", line 1843, in _execute_context\n    return self._exec_single_context(\n           ~~~~~~~~~~~~~~~~~~~~~~~~~^\n        dialect, context, statement, parameters\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\engine\\base.py\", line 1983, in _exec_single_context\n    self._handle_dbapi_exception(\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^\n        e, str_statement, effective_parameters, cursor, context\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\engine\\base.py\", line 2352, in _handle_dbapi_exception\n    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\engine\\base.py\", line 1964, in _exec_single_context\n    self.dialect.do_execute(\n    ~~~~~~~~~~~~~~~~~~~~~~~^\n        cursor, str_statement, effective_parameters, context\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"D:\\\u041f\u0440\u043e\u0435\u043a\u0442\u044b\\Threads_Hack\\venv\\Lib\\site-packages\\sqlalchemy\\engine\\default.py\", line 945, in do_execute\n    cursor.execute(statement, parameters)\n    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^\nsqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) insert or update on table \"post_table\" violates foreign key constraint \"fk_post_post_type\"\nDETAIL:  Key (post_type_id)=(0) is not present in table \"post_type_table\".\n\n[SQL: INSERT INTO post_table (post_id, content, child_id, user_id, media_link, creation_date, views_count, post_type_id) VALUES (%(post_id)s, %(content)s, %(child_id)s, %(user_id)s, %(media_link)s, %(creation_date)s, %(views_count)s, %(post_type_id)s)]\n[parameters: {'post_id': 4, 'content': '\u041f\u0440\u0438\u0432\u0435\u0435\u0435\u0435\u0435\u0435\u0435\u0442', 'child_id': None, 'user_id': 0, 'media_link': 'string', 'creation_date': datetime.date(2025, 4, 12), 'views_count': 0, 'post_type_id': 0}]\n(Background on this error at: https://sqlalche.me/e/20/gkpj)\n"
}